Tests/UnitTests/Plugins/DD4hep/DD4hepCylinderLayerStructureTests.cpp

```diff
@@ -0,0 +1,253 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Plugins/DD4hep/DD4hepDetectorSurfaceFactory.hpp"
+#include "Acts/Plugins/DD4hep/DD4hepLayerStructure.hpp"
+#include "Acts/Surfaces/SurfaceBounds.hpp"
+#include "Acts/Tests/CommonHelpers/CylindricalTrackingGeometry.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <fstream>
+#include <string>
+
+#include <DD4hep/DetElement.h>
+#include <DD4hep/DetFactoryHelper.h>
+#include <DD4hep/Detector.h>
+#include <XML/Utilities.h>
+#include <XMLFragments.hpp>
+
+#include "DD4hepTestsHelper.hpp"
+
+Acts::GeometryContext tContext;
+Acts::Test::CylindricalTrackingGeometry cGeometry =
+    Acts::Test::CylindricalTrackingGeometry(tContext);
+
+const char* beampipe_head_xml =
```

Comment:
If only we had `std::embed` 😓

Reply:
We just need the project survive until then.